framework/src/main/java/org/checkerframework/framework/util/defaults/QualifierDefaults.java

```diff
@@ -1010,6 +1010,24 @@ public Void scan(@FindDistinct AnnotatedTypeMirror t, AnnotationMirror qual) {
             // Some defaults only apply to the top level type.
             boolean isTopLevelType = t == outer.type;
             switch (outer.location) {
+                case TYPE:
+                    if (outer.scope != null && outer.scope.getKind().isClass() && isTopLevelType) {
```

Comment:
Also, another question is should we apply the class level default only for toplevel? Or nested classes as well?